Sources/Plasma/PubUtilLib/plProgressMgr/plProgressMgr.cpp

```diff
@@ -80,9 +81,13 @@ plProgressMgr::plProgressMgr()
     fCallbackProc = nullptr;
     fCurrentStaticText = kNone;
 
-    // Fill array with pre-computed loading frame IDs
-    for (int i=0; i < LOADING_RES_COUNT; i++)
-        fImageRotation[i] = ST::format("xLoading_Linking.{02}.png", i);
+    // Find linking-animation frame IDs and store the sorted list
+    std::regex re("xLoading_Linking\\.[0-9]+?\\.png");
+    for (auto name : plClientResMgr::Instance().getResourceNames()) {
```

Comment:
I believe that this makes an unnecessary copy of the string. We may prefer `const auto&`